lightning/src/util/time.rs

```diff
@@ -125,6 +125,27 @@ pub mod tests {
 		}
 	}
 
+	#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+	pub struct MockInstant(Duration);
```

Comment:
Everything is typed for concrete `Instant`s. I can change to a `Time` trait bound instead and then use `SinceEpoch` in tests. Is that kind of what you had in mind?

Reply:
Just saying that instead of needing `MockInstant`, do the following in `gossip.rs`:

```rust
#[cfg(all(test, feature = "std"))]
use util::time::{Time, tests::SinceEpoch as Instant};
```

You'll also need `use util::time::Time;` in places where `SinceEpoch` is used in tests since otherwise `now()` will not be available.

Reply:
Oh, right got it :man_facepalming: 
Yip, should work fine.